sdk/storage/azure_storage_blob/src/models/container_properties.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::{
+    headers::{
+        FromHeaders, HeaderName, Headers, ETAG, HAS_IMMUTABILITY_POLICY, HAS_LEGAL_HOLD,
+        LEASE_STATE, LEASE_STATUS,
+    },
+    Error, Etag, LeaseStatus,
+};
+use typespec_client_core::fmt::SafeDebug;
+
+use crate::models::LeaseState;
+
+pub const LAST_MODIFIED: HeaderName = HeaderName::from_static("last-modified");
+pub const IMMUTABLE_STORAGE_WITH_VERSIONING_ENABLED: HeaderName =
+    HeaderName::from_static("x-ms-immutable-storage-with-versioning-enabled");
+
+/// Properties of an Azure Storage container.
+///
+#[derive(Clone, Default, SafeDebug)]
+pub struct ContainerProperties {
+    pub last_modified: Option<String>,
```

Comment:
Why is this a string? We have date-time parsers for ISO 8601, RFC 3339, and one other RFC I forget off hand. This is not type-safe.

Reply:
Hi Heath, this hand-written model is largely cribbed from the implementation of [blob_properties.rs](https://github.com/Azure/azure-sdk-for-rust/blob/main/sdk/storage/azure_storage_blob/src/models/blob_properties.rs#L26), which also represents things i.e. `last_modified` as `Option<String>`. This is because these are stop-gap implementations since this PR consumes the `v0.9` emitter version (which does _not_ provide the serialized response header). 

This will completely change and we will no longer utilize the handwritten model once this PR is merged and we can look to start introducing the v11+ emitter changes.

Reply:
As you said: it's hand-written. Why not do it right the first time? It's straight forward because we have the helpers that do all the heavy lifting.

Reply:
To clarify, this was due to bad authoring in the tsp. It's been fixed in https://github.com/Azure/azure-rest-api-specs/commit/1530d964ce2489aae004030bead632ffbfc4ec28.